### PR TITLE
Remove Tracer packet as a concept

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
- I quietly rugged the metrics reported by banking stage for tracer packets
- No one complained...which means no one uses these

#### Summary of Changes
- Remove Tracer Packet concept at all levels

Fixes #

### CODE REVIEW

**Comment by apfitzge** [comment]
@ryoqun could affect you - this will break backward compatibility with banking-trace files.

**Comment by ryoqun** [comment]
> @ryoqun could affect you - this will break backward compatibility with banking-trace files.

I'm okay with this new abi incompatibility.

**Comment by apfitzge** [comment]
> I'm okay with this new abi incompatibility.

Realizing I should probably also add a note to the change log about this.
Best if operators delete/move their old banking-trace when upgrading to version supporting this, because otherwise the files will be a mixture of both. If they don't would just make the files useless for the few days/hours it takes to roll over.

File: core/src/sigverify.rs [hunk 1]
```diff
@@ -88,49 +47,24 @@ impl SigVerifier for TransactionSigVerifier {
     #[inline(always)]
```
Comment (bw-solana) [inline core/src/sigverify.rs]: can we just take the default trait implementation for these (besides send of course)?

File: sdk/packet/src/lib.rs [hunk 2]
```diff
@@ -52,9 +52,10 @@ bitflags! {
         const FORWARDED      = 0b0000_0010;
         const REPAIR         = 0b0000_0100;
         const SIMPLE_VOTE_TX = 0b0000_1000;
-        const TRACER_PACKET  = 0b0001_0000;
         // Previously used - this can now be re-used for something else.
-        const UNUSED = 0b0010_0000;
+        const UNUSED_0  = 0b0001_0000;
+        // Previously used - this can now be re-used for something else.
+        const UNUSED_1 = 0b0010_0000;
```
Comment (bw-solana) [inline sdk/packet/src/lib.rs:58]: Do we need like a cool off period before this is re-used? So that old versions don't misinterpret as tracer packet

**Review by bw-solana — COMMENTED** [review]
LGTM overall. Couple minor things to consider

File: sdk/packet/src/lib.rs (hunk 2 above)
Comment (apfitzge) [reply sdk/packet/src/lib.rs:58]: Not sure it's useful.
The only place these will show up in a public setting are in banking-trace data.

File: core/src/sigverify.rs (hunk 1 above)
Comment (apfitzge) [reply core/src/sigverify.rs]: Even more dead lines!

Comment (apfitzge) [reply core/src/sigverify.rs]: Used the default implementations: cdaa1b02320be1497fc6045bd6c672f7e9a10797

Then realized, now that THIS isn't implementing those functions...nothing is.
fc49d453b17dd52270748898bef416add54f7a30, fb0ab3242c0491f1e8e64528f8f8281e5d474dca, 9a23f07ebd7ce2f5cf81aafd190816cfcc382a37
Let us remove an additional ~100 lines of unused code.

Thanks for pointing that out @bw-solana!

**Review by bw-solana — APPROVED** [review]
LGTM

**Pull request merged**